content/Projects/Observability/kube-rbac-proxy.md

```diff
@@ -0,0 +1,52 @@
+# kube-rbac-proxy
+
+`kube-rbac-proxy`, as the name suggests, is a proxy that sits in front of a workload and performs authentication 
+and authorization on its behalf, to make sure the requests passing through are the ones that it expects to.
+
+## Workflow
+
+The purpose of `kube-rbac-proxy` is to distinguish between internal calls made by same or different user(s) (or 
+service account(s)) to endpoint(s) and protect them from unauthenticated or unauthorized resource access based on 
+the Bearer tokens or the RBACs they hold, respectively.
+
+- [**Authentication**]: Both [OIDC JWT] and [delegated (header-based)] authentication rely on Bearer tokens to 
+  authenticate the requests. The tokens are validated against the OIDC provider or the Kubernetes API server, 
+  respectively. The proxy can also be configured to [skip the authentication step] altogether, in which case it will 
+  not validate the tokens and will assume that the requests are authenticated. This token represents the identity of 
+  the user or service account that is making the request. A [`TokenReview` request is created], and includes the bearer 
+  token that the client provided, which is then sent to the API server. The API server verifies the authenticity of 
+  the token and responds with the details of the authenticated party, if the operation was successful.
+
+[skip the authentication step]: https://github.com/rexagod/kube-rbac-proxy/blob/4c4a18fa7dd5d8069c0aa3a55028e06631621a52/pkg/authn/delegating.go#L53
+[OIDC JWT]: https://github.com/rexagod/kube-rbac-proxy/blob/52e49fbdb75e009db4d02e3986e51fdba0526378/pkg/authn/oidc.go#L45-L63
+[delegated (header-based)]: https://github.com/kubernetes/apiserver/blob/master/pkg/authentication/authenticatorfactory/delegating.go#L102-L112
+[**Authentication**]: https://github.com/rexagod/kube-rbac-proxy/blob/74181c75b8b6fbcde7eff1ca5fae353faac5cfae/pkg/authn/config.go#L33-L39
+[`TokenReview` request is created]: https://github.com/kubernetes/apiserver/blob/21bbcb57c672531fe8c431e1035405f9a4b061de/plugin/pkg/authenticator/token/webhook/webhook.go#L51-L53
+
+- [**Authorization**]: Once authentication is done, `kube-rbac-proxy` must then decide whether to allow the user's 
```

Comment:
Static authorization is also used for /metrics endpoint to avoid another request to the API server (and better resiliency).